### ISSUE TITLE
Snappy.java's  decodeLiteral method implemention

### ISSUE THREAD

**Issue by johnsonma** [netty/netty#4699]
```java
    private static int decodeLiteral(byte tag, ByteBuf in, ByteBuf out) {
        in.markReaderIndex();
        int length;
        switch(tag >> 2 & 0x3F) {
        case 60:
            if (!in.isReadable()) {
                return NOT_ENOUGH_INPUT;
            }
            length = in.readUnsignedByte();
            break;
        case 61:
            if (in.readableBytes() < 2) {
                return NOT_ENOUGH_INPUT;
            }
            length = ByteBufUtil.swapShort(in.readShort());
            break;
        case 62:
            if (in.readableBytes() < 3) {
                return NOT_ENOUGH_INPUT;
            }
            length = ByteBufUtil.swapMedium(in.readUnsignedMedium());
            break;
        case 64:
            if (in.readableBytes() < 4) {
                return NOT_ENOUGH_INPUT;
            }
            length = ByteBufUtil.swapInt(in.readInt());
            break;
        default:
            length = tag >> 2 & 0x3F;
        }
        length += 1;

        if (in.readableBytes() < length) {
            in.resetReaderIndex();
            return NOT_ENOUGH_INPUT;
        }

        out.writeBytes(in, length);
        return length;
    }
```
I was wondering to know why there's a 'case 64' here. Should 'tag >> 2 & 0x3F' never will be equal to '64'?

**Comment by normanmaurer** [comment]
@lw346 could you please comment ?

**Comment by zsxwing** [comment]
I guess he meant any number `&` with `0x3F` (63) must be less than 64.

**Comment by windie** [comment]
Fixed in #4731
